tlslite/utils/python_tripledes.py

```diff
@@ -0,0 +1,579 @@
+#################################################
+#               Documentation                   #
+#################################################
+
+# Author:   Todd Whiteman
+# Date:     16th March, 2009
+# Verion:   2.0.0
+# License:  Public Domain - free to do as you wish
+# Homepage: http://twhiteman.netfirms.com/Des.html
+#
+# Modified by: Adam Varga, 2018
+#
+# Triple DES class is implemented by utilising the DES base. Triple DES
+# is DES-EDE3 with a 24 byte key, or DES-EDE2 with a 16 byte key.
+"""
+A pure python implementation of the DES and TRIPLE DES
+encryption algorithms using CBC mode.
+
+Class initialization
+--------------------
+pyDes.Des(key, [IV])
+pyDes.Triple_des(key, [IV])
+
+key -> Bytes containing the encryption key. 8 bytes for DES, 16 or 24 bytes
+       for Triple DES
+IV  -> Optional Initial Value bytes, must be supplied if using CBC mode.
+       Length must be 8 bytes.
+
+Common methods
+--------------
+encrypt(data)
+decrypt(data)
+
+data   -> Bytes to be encrypted/decrypted
+"""
+
+import sys
+# from tripledes import _baseDes
+# from tripledes import DES
+# from tripledes import TripleDES
+
+
+# PY_VER is used to handle Python2 and Python3 differences.
+PY_VER = sys.version_info[0]
+
+# the new() method to operate this cipher
+def new(key, IV=None):
+    return Triple_des(key, IV)
+
+# The base class shared by des and triple des.
+class _baseDes(object):
+    def __init__(self, IV=None):
+        if IV:
+            IV = self._guardAgainstUnicode(IV)
+        self.block_size = 8
+
+        # Sanity checking of arguments.
+        if IV and len(IV) != self.block_size:
+            raise ValueError("Invalid Initial Value (IV), must be \
+                a multiple of " + str(self.block_size) + " bytes")
```

Comment:
this will result in an error _with_ a newline, also `format()` operator should be used for string interpolation:
```
            raise ValueError("Invalid Initial Value (IV), must be "
                             "a multiple of {0} bytes".format(self.block_size))
```